docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -271,16 +271,66 @@ This may cause that incorrect results are returned for the query.
    >        --id $ID \         
    > done
    >```
-  
-5) Copy the blocks from the intermediate bucket to the Mimir bucket
+ 
+5) Remove external labels from meta.json
+   
+   Mimir compactor will not be able to compact the blocks with external labels in their meta.json. Therefore these external labels have to be  removed before copying them to the Mimir bucket.
+   
+   ```bash
+   # Create a folder named "orig".
+   mkdir -p orig
+   
+   # Copy the meta.json file from all the blocks to orig folder in your local machine excluding the index and chunk folders.
+   gsutil -m rsync -r -x '.*(index|chunks|debug).*' "gs://{Intermediate Mimir bucket}/" orig/
+   ```
+   
+   Use the below script to remove the labels from the meta.json.
+   
+   ```bash
+   #!/bin/bash
+   # Description: Update Thanos external labels
+   # Create a folder called output
+   mkdir -p output
+   
+   # Get all meta.json file under the orig folder
+   find orig -name 'meta.json' |
+   while read orig; do
+     block="$(echo "${orig}" | cut -f2 -d'/')"
+     tmpfile=$(mktemp)
+   
+     # use jq to delete the labels from the meta.sjon file
+     jq 'del(.thanos.labels.<LABEL-KEY-1>, .thanos.labels.<LABEL-KEY-2), .thanos.labels.<LABEL-KEY-3>)' "${orig}" > "${tmpfile}"
```

Comment:
We can delete entire `.thanos.labels` section, not individual just labels.

Can we simplify the script? For example this would work as well:

```sh
for FILE in $(gsutil ls 'gs://<bucket>/*/meta.json'); do
    # Back up original meta.json file.
    gsutil cp $FILE $FILE.orig
    # Remove Thanos labels section from meta.json file.
    gsutil cat $FILE | jq 'del(.thanos.labels)' | gsutil cp - $FILE
done
```

Reply:
The reason why I avoided doing these changes from the gcloud command line is that when I do it from the local machine. I would have more control over the changes taking place. We can delete the entire labels section but do we need to simplify the script to this level as it's not showing any output until it's done if we have a lot of blocks in the gcs bucket.